src/test/java/org/symphonyoss/symphony/messageml/Mml2Pml2Pml.java

```diff
@@ -0,0 +1,139 @@
+package org.symphonyoss.symphony.messageml;
+
+import static org.apache.commons.io.FilenameUtils.getBaseName;
+import static org.apache.commons.io.FilenameUtils.isExtension;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.symphonyoss.symphony.messageml.elements.MessageML;
+import org.symphonyoss.symphony.messageml.util.IDataProvider;
+import org.symphonyoss.symphony.messageml.util.TestDataProvider;
+import org.w3c.dom.Node;
+import org.w3c.dom.bootstrap.DOMImplementationRegistry;
+import org.w3c.dom.ls.DOMImplementationLS;
+import org.w3c.dom.ls.LSSerializer;
+import org.xml.sax.InputSource;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+
+/**
+ * This test loads a set of MessageML files from classpath:/examples location and performs 2 successive transformation:
+ * <ul>
+ *   <li>MessageML -> PresentationML</li>
+ *   <li>PresentationML -> PresentationML</li>
+ * </ul>
+ * It allows to ensure that generated PresentationML is an actual MessageML valid one.
+ */
+@Slf4j
+public class Mml2Pml2Pml {
+
+  private static final String EXAMPLES = "/examples/";
+
+  @ParameterizedTest(name = "File {0}")
+  @MethodSource("examples")
+  void mml2Pml2Pml(String file, String message, String entityJson) throws Exception {
+
+    log.info("\n----\n### EXAMPLE({})", file);
+    log.info("1 # MML:\n{}\n{}{}", LINE, prettyXml(message), LINE);
+    //
+    // STEP 1 / MML -> PML
+    //
+    log.info("1 # MML -> PML:");
+    MessageMLContext context = parse(message, entityJson);
+    log.info("\n{}\n{}{}", LINE, prettyXml(context.getPresentationML()), LINE);
+    //
+    // STEP 2 / PML -> PML
+    //
+    log.info("2 # PML -> PML:");
+    context = parse(context.getPresentationML(), context.getEntityJson());
+    context.getText();
+    log.info("\n{}\n{}{}", LINE, prettyXml(context.getPresentationML()), LINE);
```

Comment:
should we actually (or did I miss it?) check that presentation ml is actually the same (is it or can it change a bit)?

Reply:
Actually for some elements, we introduce a random `id` prefix when converting MML -> PML ([here](https://github.com/finos/messageml-utils/blob/master/src/main/java/org/symphonyoss/symphony/messageml/elements/Dialog.java#L105) for instance) which prevent us from comparing the 2 PresentationMLs

Reply:
Done [here](https://github.com/finos/messageml-utils/pull/315/files#diff-878b6ef1d28806245245308ed46532eda41aa60eb7c43f9656cda16938430f47R77-R79) @symphony-youri, I used [xmlunit](https://github.com/xmlunit/xmlunit) to exclude properties with random values from XML tree comparison